### PR TITLE
Storage Mar. Release

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by Jinming-Hu** [comment]
@danieljurek @vhvb1989 There's something wrong with the pipeline, can you please take a look?

**Comment by danieljurek** [comment]
This failure appears on re-run. It's an issue from vcpkg installing `libiconv` ... This might be fixable by moving our vcpkg SHA up to a later version in `eng/vcpkg-commit.txt`

**Review by vhvb1989 — APPROVED** [review]

File: eng/vcpkg-commit.txt [hunk 1]
```diff
@@ -1 +1 @@
-fcd3a3bb53356cc954f7fa07cee2c99b253d1a49
+be5c4ef68b51142ba705f0678b45d284977de677
```
Comment (ahsonkhan) [inline eng/vcpkg-commit.txt:1]: Why did we change this?

Comment (Jinming-Hu) [reply eng/vcpkg-commit.txt:1]: To fix CI failure as suggested by @danieljurek . It worked

Comment (ahsonkhan) [reply eng/vcpkg-commit.txt:1]: Looks like we picked the recent commit from vcpkg:
https://github.com/microsoft/vcpkg/commit/be5c4ef68b51142ba705f0678b45d284977de677

image

I am not sure if it is the best idea to point to any arbitrary commit from vcpkg, but maybe that's good enough. We need to consider adding some mechanism to test the SDK, against the oldest dependencies we support, to validate things continue to work for customers.

The last time we bumped it, it was a commit from Nov.
https://github.com/Azure/azure-sdk-for-cpp/pull/3154

**That said, picking a sha right after the latest azure core release makes sense to me!**
In this case, that would be:
97113b43b3b6905a8c583900a0540471cfa88d6e
https://github.com/microsoft/vcpkg/pull/23371

Comment (ahsonkhan) [reply eng/vcpkg-commit.txt:1]: This is the testing strategy vcpkg folks suggest:
https://github.com/Azure/azure-sdk-for-cpp/pull/3154#discussion_r763468105

**Pull request merged**